bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RackawareEnsemblePlacementPolicyImpl.java

```diff
@@ -531,26 +561,40 @@ public void handleBookiesThatJoined(Set<BookieSocketAddress> joinedBookies) {
         return networkLocs;
     }
 
+    /*
+     * this method should be called in readlock scope of 'rwLock'
+     */
+    protected Set<BookieSocketAddress> addDefaultZoneBookiesIfMinNumRacksIsEnforced(
+            Set<BookieSocketAddress> excludeBookies) {
+        Set<BookieSocketAddress> comprehensiveExclusionBookiesSet;
+        if (enforceMinNumRacksPerWriteQuorum) {
+            comprehensiveExclusionBookiesSet = new HashSet<BookieSocketAddress>(excludeBookies);
```

Comment:
Looks like you have put this inside the if condition to avoid unnecessary object allocation. To serve that purpose, may be you should check if the defaultRack is empty or not before allocating new object.

Reply:
will do